packages/store-ui/src/atoms/Button/Button.tsx

```diff
@@ -2,8 +2,17 @@ import type { ButtonHTMLAttributes, ElementType } from 'react'
 import React, { forwardRef } from 'react'
 
 export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  /**
+   * Specifies the URL of the page this button click goes to.
+   */
   href?: string
```

Comment:
why our button has an href?

Reply:
I was wondering when I saw it too. @emersonlaurentino any idea? I described the prop what I thought it was, but I'm not sure.

Reply:
I think we can remove it for now

Reply:
oh, no!! The button must have the `href` prop because when the user put `as` with anchor element (a) he will want to pass the href prop but receive an error. @Gmantiqueira @tlgimenes

Reply:
But then shouldn't we be able to do something like ?
```
export interface ButtonProps<T = HTMLButtonElement> extends ButtonHTMLAttributes<T>
```

Reply:
I don't like this usage btw. If I need to redirect from a button, makes more sense to me using the onClick callback with a redirect/navigate, or wrap the button with an anchor or just use a styled anchor. 

This "as" prop can have N usage possibilities and if we are fixing this case, we need to get all of the possibilities and it doesn't makes sense to me at all.

Reply:
no @Gmantiqueira because the user wanna extend the Button UI, like a Link that looks like a button, but is a Link.

Reply:
@emersonlaurentino  Exactly, "a Link like a Button" not "a Button as a Link", that's why I don't like to look at our components that way too flexible. 

Why we're importing a Button if we want to use a Link? A Button shouldn't be responsible for href/redirect/navigation, I think his function should be mainly to have a callback on click, not to be a "base-style" for links. Then it makes more sense to me if we scope each function in your respective component. What do you think?

Reply:
I think the "as" prop makes more sense in components without a "scoped-function" like span, div, section, texts etc. But i'm a bit aware of opening doors for the "as" prop in our Buttons/Anchors/Links. I think each component needs to have their respective mainly functions only.

Reply:
I agree with @Gmantiqueira . If I'm reading a code from a store and I read the following code:
```jsx
const Component = () => <Button />
```

I will tend to think this will be rendered to the following html `<button />`

However, if I read the following code:

```
const Component = <Price />
```

I dont really know what's going to be the generated html, So in this case I agree having an `as` prop.

Reply:
well, okay. Just remove the prop `as` from the button.